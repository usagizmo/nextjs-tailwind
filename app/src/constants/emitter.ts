export const EMITTER = {
  scrollTop: 'SCROLL_TOP',
  animatedScrollTop: 'ANIMATED_SCROLL_TOP',
  bodyHeight: 'BODY_HEIGHT',
  bodyOverflowY: 'BODY_OVERFLOW_Y',
  mouseMove: 'MOUSE_MOVE',
}
