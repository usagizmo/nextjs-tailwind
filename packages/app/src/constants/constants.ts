export const TWITTER_NAME = 'usagizmo'
