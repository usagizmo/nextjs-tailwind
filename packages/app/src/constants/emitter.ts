export const EMITTER = {
  animatedScrollTop: 'ANIMATED_SCROLL_TOP',
  animationFrame: 'ANIMATION_FRAME',
  bodyHeight: 'BODY_HEIGHT',
  mouseDown: 'MOUSE_DOWN',
  mouseMove: 'MOUSE_MOVE',
  mouseUp: 'MOUSE_UP',
  scrollTop: 'SCROLL_TOP',
}
