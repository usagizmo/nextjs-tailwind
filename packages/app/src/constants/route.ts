export const ROUTE = {
  HOME: '/',
  DETAIL_ID: '/detail/[id]',
}
