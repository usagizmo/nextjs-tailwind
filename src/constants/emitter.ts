export const EMITTER = {
  scrollY: 'SCROLL_Y',
  animatedScrollY: 'ANIMATED_SCROLL_Y',
  windowSize: 'WINDOW_SIZE',
}
