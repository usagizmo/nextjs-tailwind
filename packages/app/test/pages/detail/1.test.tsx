import React from 'react'
import { render } from '../../testUtils'
import DetailPage from '../../../src/pages/detail/[id]'
import { ROUTE } from '../../../src/constants/route'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

describe('Home page', () => {
  it('matches snapshot', () => {
    useRouter.mockImplementationOnce(() => ({
      pathname: ROUTE.DETAIL_ID,
      query: { id: 1 },
    }))
    const { asFragment } = render(<DetailPage />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
