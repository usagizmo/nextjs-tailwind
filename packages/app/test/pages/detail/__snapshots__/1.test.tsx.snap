// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="h-full flex flex-col items-center"
  >
    <ul
      class="flex"
    >
      <li
        class="px-4 text-gray-500 hover:text-gray-900"
      >
        <a
          href="/"
        >
          HOME
        </a>
      </li>
      <li
        class="px-4 text-gray-500 hover:text-gray-900"
      >
        <a
          href="/detail/1"
        >
          detail/1
        </a>
      </li>
      <li
        class="px-4 text-gray-500 hover:text-gray-900"
      >
        <a
          href="/detail/2"
        >
          detail/2
        </a>
      </li>
    </ul>
    <p
      class="mt-20 text-xl"
    >
      <code>
        detail/[id].tsx - query: { id: 1 }
      </code>
    </p>
  </main>
</DocumentFragment>
`;
